parcels/kernel.py

```diff
@@ -390,7 +301,13 @@ def evaluate_particle(self, p, endtime):
             except KeyError:
                 if sign_dt * (endtime - p.time_nextloop) <= p.dt:
                     p.dt = sign_dt * (endtime - p.time_nextloop)
-            res = self._pyfunc(p, self._fieldset, p.time_nextloop)
+            res = None
+            for f in self._pyfuncs:
+                res_tmp = f(p, self._fieldset, p.time_nextloop)
+                if res_tmp is not None:  # TODO v4: Remove once all kernels return StatusCode
+                    res = res_tmp
```

Comment:
In the JIT version of v3, we kept the “highest” statuscode, with the idea that this was the most important

Reply:
What you're refering to is in JIT mode during interpolation. This is different from the particle status code that is returned after each kernel (which is what this part of the code is handling).

Leaving as is.